prov/efa/src/efa_mr.c

```diff
@@ -176,32 +205,121 @@ struct fi_ops_mr efa_domain_mr_cache_ops = {
 	.regattr = efa_mr_cache_regattr,
 };
 
+static int efa_mr_release(struct efa_mr *mr)
+{
+	struct efa_domain *efa_domain;
+	int ret = 0;
+	int err;
+
+	efa_domain = mr->domain;
+	err = -ibv_dereg_mr(mr->mr);
+	if (!err) {
+		EFA_WARN(FI_LOG_MR,
+			"Unable to close MR\n");
+		ret = err;
+	}
+	err = ofi_mr_map_remove(&efa_domain->util_domain.mr_map,
+				mr->mr_fid.key);
+	if (err && err != -FI_ENOKEY) {
```

Comment:
Same comment as the one in efa_mr_map_insert_shm_reg. We should not ignore FI_ENOKEY any more.

Reply:
I agree with zhngaj here, ibv_reg_mr should not return same key if the key is being actively used. Please double check.